x/gamm/pool-models/stableswap/amm_test.go

```diff
@@ -867,16 +868,20 @@ func TestCalcSingleAssetJoinShares(t *testing.T) {
 
 	for name, tc := range tests {
 		t.Run(name, func(t *testing.T) {
-			ctx := sdk.Context{}
+			// Init suite for each test.
+			ctx := sdk.Context{}.WithGasMeter(sdk.NewGasMeter(1_000_000_000))
 			p := poolStructFromAssets(tc.poolAssets, tc.scalingFactors)
 
-			shares, err := p.calcSingleAssetJoinShares(tc.tokenIn, tc.swapFee)
+			shares, err := p.calcSingleAssetJoinShares(ctx, tc.tokenIn, tc.swapFee)
 			require.NoError(t, err, "test: %s", name)
 
 			p.updatePoolForJoin(sdk.Coins{tc.tokenIn}, shares)
+			prevGasConsumed := ctx.GasMeter().GasConsumed()
 			exitTokens, err := p.ExitPool(ctx, shares, sdk.ZeroDec())
 			require.NoError(t, err, "test: %s", name)
-
+			gasConsumedForSwap := ctx.GasMeter().GasConsumed() - prevGasConsumed
```

Comment:
is it possible to just do something like this:
```
gasConsumedForSwap := ctx.GasMeter().GasConsumed()
```

w/o subtracting `prevGasConsumed`. It seems to me that it is 0 every time, because we initialize a new `ctx` for every test. 

Or, even better I think, we could define a `ctx` outside of testing loop